project-base/src/Shopsys/ShopBundle/DataFixtures/Demo/UserDataFixture.php

```diff
@@ -11,75 +11,390 @@
         CustomerFacade $customerFacade,
-        UserDataFixtureLoader $loaderService,
         Generator $faker,
         EntityManagerInterface $em,
-        HashGenerator $hashGenerator
+        HashGenerator $hashGenerator,
+        Domain $domain,
+        CustomerDataFactoryInterface $customerDataFactory,
+        UserDataFactoryInterface $userDataFactory,
+        BillingAddressDataFactoryInterface $billingAddressDataFactory,
+        DeliveryAddressDataFactoryInterface $deliveryAddressDataFactory
     ) {
         $this->customerFacade = $customerFacade;
-        $this->loaderService = $loaderService;
         $this->faker = $faker;
         $this->em = $em;
         $this->hashGenerator = $hashGenerator;
+        $this->domain = $domain;
+        $this->customerDataFactory = $customerDataFactory;
+        $this->userDataFactory = $userDataFactory;
+        $this->billingAddressDataFactory = $billingAddressDataFactory;
+        $this->deliveryAddressDataFactory = $deliveryAddressDataFactory;
     }
 
     /**
      * @param \Doctrine\Common\Persistence\ObjectManager $manager
      */
     public function load(ObjectManager $manager)
     {
-        $countries = [
-            $this->getReference(CountryDataFixture::COUNTRY_CZECH_REPUBLIC),
-            $this->getReference(CountryDataFixture::COUNTRY_SLOVAKIA),
-        ];
-        $this->loaderService->injectReferences($countries);
-
-        $customersData = $this->loaderService->getCustomersDataByDomainId(Domain::FIRST_DOMAIN_ID);
-
-        foreach ($customersData as $customerData) {
-            $customerData->userData->createdAt = $this->faker->dateTimeBetween('-2 week', 'now');
+        foreach ($this->domain->getAll() as $domainConfig) {
+            $domainId = $domainConfig->getId();
+            if ($domainId === 2) {
+                $customersData = $this->getCustomersDataForSecondDomain();
+            } else {
+                $customersData = $this->getCustomersDataForOtherThanSecondDomain($domainId);
+            }
 
-            $customer = $this->customerFacade->create($customerData);
+            foreach ($customersData as $customerData) {
+                $customerData->userData->createdAt = $this->faker->dateTimeBetween('-1 week', 'now');
 
-            if ($customer->getId() === 1) {
-                $this->resetPassword($customer);
-                $this->addReference(self::USER_WITH_RESET_PASSWORD_HASH, $customer);
+                $customer = $this->customerFacade->create($customerData);
+                if ($customer->getId() === 1) {
+                    $this->resetPassword($customer);
+                    $this->addReference(self::USER_WITH_RESET_PASSWORD_HASH, $customer);
+                }
             }
         }
     }
 
+    /**
+     * @param int $domainId
+     * @return \Shopsys\FrameworkBundle\Model\Customer\CustomerData[]
+     */
+    protected function getCustomersDataForOtherThanSecondDomain(int $domainId): array
```

Comment:
I have still the same problem :smile: why isn't it call somehow else like default? Because if you would like to have different data for 3. domain you'll need to rename this to `getCustomersDataForOtherThanSecondAndThirdDomain` for it to make sense..

Reply:
fixed in c199c3552ae11aed69b599ab1ba187a6a3984215